rep-2004.rst

```diff
@@ -72,137 +78,123 @@ For example, it may be the case the tool which launches and verifies a ROS graph
+For other languages the details will be different, but the important thing is that the public API be obviously documented, and the public API adheres to a documented and tested API and ABI stability policy, as described in the version policy.
 
 Feature Documentation
-"""""""""""""""""""""
+~~~~~~~~~~~~~~~~~~~~~
 
 For each feature provided by the public API of the package, or by a tool in the package, there must be corresponding user documentation.
 The term "feature", and the scope of the documentation, is intentionally vague because it's difficult to quantitatively measure this metric.
-However, the spirit of this requirement is that, for a 'Level 1' quality package, all of the things a user might do with the package needs at least basic documentation or a snippet of code as an example on how to use it.
+In general, for a 'Level 1' quality package, all of the things a user might do with the package need at least basic documentation or a snippet of code as an example on how to use it.
 The `roscpp Overview <https://wiki.ros.org/roscpp/Overview>`_ from the ROS 1 wiki is a good example of this kind of documentation.
 
+Quality Declaration and Claim
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each package claiming a quality level should include a "quality declaration" document.
+This declaration should include a claimed quality level and then should have a section for each of the requirements in that claimed quality level justifying how the package meets each of those requirements.
+
+Sometimes the justification will be a link to a policy documented in the package itself or it may link to a common policy used by a group of packages.
+If there is additional evidence that these policies are being followed, that should be included as well, e.g. a link to the coverage statistics for the package to show that coverage is being tracked and maintained.
+Other times, justification will be an explanation as to why a requirement was not met or does not apply, e.g. if performance tests do not make sense for the package in question, it should be satisfactorily explained.
+
+There is no enforcement or checking of these claims, but instead it's just sufficient to present this information to potential users.
+If the users feel that the justifications are insufficient or incorrect, they can open issues against the repository and resolve it with the maintainers.
+
+There should be one or more communal lists of 'Level 1' (and maybe 'Level 2' or 'Level 3') quality level packages.
+These lists should be modified via change requests (maybe a text document in a repository) so that there can be peer review.
+This document will not prescribe how or where these lists should be hosted, but one thought is that the list could live on the main ROS 2 documentation website.
```

Comment:
@wjwwood should I just create this page now and link it here? it would be better for the REP but presumably wouldn't have any content yet/for a while

Reply:
Not sure. I hesitate because I don't know where it should live (in the `ros2/ros_documentation` repository or a separate repository similar to how REPs are in their own `ros2/rep` repository). It's possible a REP which is continually updated might be a good way to do it too, sort of how we update REP-2000 for each new ROS 2 release. I'll mention this in the next ROS 2 meeting.